megamek/src/megamek/client/bot/princess/MinefieldUtil.java

```diff
@@ -0,0 +1,99 @@
+/*
+* MegaMek -
+* Copyright (C) 2021 The MegaMek Team
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the GNU General Public License as published by the Free Software
+* Foundation; either version 2 of the License, or (at your option) any later
+* version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+* details.
+*/
+
+package megamek.client.bot.princess;
+
+import megamek.common.Compute;
+import megamek.common.Entity;
+import megamek.common.EntityMovementMode;
+import megamek.common.Mech;
+import megamek.common.Minefield;
+import megamek.common.MovePath;
+import megamek.common.MoveStep;
+
+/**
+ * This class contains logic to evaluate the damage a unit could sustain from
+ * moving a long a move path containing minefields
+ * @author NickAragua
+ */
+public class MinefieldUtil {
+    /**
+     * Calculate how much damage we'll take from stepping on mines over  a particular path
+     */
+    public static double checkPathForMinefieldHazards(MovePath path) {
+        double hazardAccumulator = 0;
+                
+        for (MoveStep step : path.getStepVector()) {
+            hazardAccumulator += calcMinefieldHazardForHex(step, path.getEntity(),
+                    path.isJumping(), step.equals(path.getLastStep())); 
+        }
+        
+        //TODO: Teach bot to activate minesweepers
+        
+        return hazardAccumulator;
+    }
+    
+    /**
+     * Calculate how much damage we'll take from stepping on mines in a particular hex
+     */
+    public static double calcMinefieldHazardForHex(MoveStep step, Entity movingUnit, 
+            boolean isJumping, boolean lastStep) {
+        // if we're not actually taking a step, no minefield hazard
+        if ((step == null) || !step.getType().entersNewHex()) {
+            return 0;
+        }
+        
+        // if our movement mode does not result in minefield detonation, no minefield hazard
+        if (!movingUnit.getMovementMode().detonatesGroundMinefields()) {
+            return 0;
+        }
+        
+        double hazardAccumulator = 0;
+        // hovercraft and WIGEs grinding along the ground detonate minefields on a 12
+        boolean hoverMovement = movingUnit.getMovementMode() == EntityMovementMode.HOVER ||
+                (movingUnit.getMovementMode() == EntityMovementMode.WIGE && movingUnit.getElevation() == 0);
+        double hoverMovementMultiplier = hoverMovement ?
+                Compute.oddsAbove(Minefield.HOVER_WIGE_DETONATION_TARGET) : 1;
+        
+        // only mechs interact with vibrabombs
+        boolean unitIsMech = movingUnit instanceof Mech;
+        
+        for (Minefield minefield : movingUnit.getGame().getMinefields(step.getPosition())) {
```

Comment:
Per your comment, this is only coming back with details both humans and bots would have right? No leg up w.r.t. type?

Reply:
Correct, it's just what the player would see if they were looking at the board.